modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -1802,18 +1805,16 @@ public void postProcessResponseWithProperty(CodegenResponse response, CodegenPro
             return;
         }
 
-        // the response data types should not contain a bean validation annotation.
-        if (property.dataType.contains("@")) {
-            property.dataType = removeAnnotations(property.dataType);
-        }
-        // the response data types should not contain a bean validation annotation.
-        if (response.dataType.contains("@")) {
-            response.dataType = removeAnnotations(response.dataType);
-        }
+        // the response data types should not contain bean validation annotations.
+        property.dataType = removeAnnotations(property.dataType);
+        response.dataType = removeAnnotations(response.dataType);
     }
 
-    private String removeAnnotations(String type) {
-        return type.replaceAll("(?:(?i)@[a-z0-9]*+([(].*[)]|\\s*))*+", "");
+    private String removeAnnotations(String dataType) {
```

Comment:
@jpraet thanks for the PR.

the only suggestion I've is adding s docstring explaining what this function does with an example or 2 before and after applying this function (essentially explaining what the regular express does)?

Reply:
added